Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -776,7 +776,19 @@ AllowTrustedTypePolicy ContentSecurityPolicy::allowTrustedTypesPolicy(const Stri
             "Refused to create a TrustedTypePolicy named '"_s, value, "' because it violates the following Content Security Policy directive: \""_s, violatedDirective.text(), '"');
         reportViolation(violatedDirective, "trusted-types-policy"_s, consoleMessage, sourceURL, StringView(value), sourcePosition);
     };
-    auto isAllowed = allPoliciesAllow(WTFMove(handleViolatedDirective), &ContentSecurityPolicyDirectiveList::violatedDirectiveForTrustedTypesPolicy, value, isDuplicate, details);
+
+    bool isAllowed = true;
+    for (auto& policy : m_policies) {
+        AllowTrustedTypePolicy perPolicyDetails = AllowTrustedTypePolicy::Allowed;
+        if (const ContentSecurityPolicyDirective* violatedDirective = policy.get()->violatedDirectiveForTrustedTypesPolicy(value, isDuplicate, perPolicyDetails)) {
+            if (!violatedDirective->directiveList().isReportOnly())
+                isAllowed = false;
+            // If we already have a violation, we should use the details from the first policy that violated the directive.
+            if (!isAllowed && details == AllowTrustedTypePolicy::Allowed)
+                details = perPolicyDetails;
+            handleViolatedDirective(*violatedDirective);
```

Comment:
Do we need to go through all directives if one was violated? (maybe we do, I don't really know this spec well enough)

Reply:
For the spec it does loop through all of them and report multiple violations: https://w3c.github.io/trusted-types/dist/spec/#should-block-create-policy

I don't know if chrome actually does that, but for now I'm going to implement against the spec and we can adjust as necessary. If there aren't already more tests for that will be added.

Reply:
Can you add a link to the spec on top of this function or wherever appropriate?

Reply:
Done.